### ISSUE TITLE
Accessibility Review of Claims & Appeals - 526 Wizard updates

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22569]
# Accessibility Review of Claims & Appeals - 526 Wizard updates

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Angela and Josh

**Comment by joshkimux** [comment]
I have no accessibility feedback to add to this ticket beyond what has already been documented in [the ticket to retire the current "magic" (dynamic) version of the wizard](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/399). 

The most relevant of the issues documented there would likely be around feedback, specifically around dates.

Previously, (from my not so good memory) I believe an error message would be thrown if the user did not complete all of the fields in the right order. For example, working backwards from year -> day -> month would draw an error upon filling in year. That was undesired, as it was throwing an error to the user for something that wasn't an error yet. 

On the flip side, if no errors are thrown until all fields are completed, I wonder if there exists a case where a user may interpret "anticipated date" as being flexible. For example, what if they were to fill in just month and year but not the date?

https://user-images.githubusercontent.com/14154792/123815726-b6faa000-d8c4-11eb-92c1-1fbafc148949.mov

Are all of the date fields required to receive feedback? What if a user accidentally skips over a field? (e.g. a blind mobile screen reader user who swipes past the day field accidentally)

A more elegant fix would be to include a submit button-- but in the meantime-- are proactive steps needed to prevent these issues and can we provide them?

**Comment by Mottie** [comment]
> an error message would be thrown if the user did not complete all of the fields in the right order

The "Day" select options are dependent on the "Month" selection, so there won't be any choices  until a month is selected. It's not ideal, but this is the global date selector React component used across the site.

> user may interpret "anticipated date" as being flexible

The anticipated date isn't fixed until the active service member gets closer to discharge, but we do need an exact date entered because they wouldn't be qualified for BDD (Benefits Delivery at Discharge) if they are not within the 90 to 180 days before that anticipated date. So we do need it to be as exact as possible, meaning we need the full date.

> Are all of the date fields required to receive feedback?

Yes. If the incomplete date fields are blurred before a full date is entered, there is no feedback. It's done that way so the error messages don't show up while navigating between fields. Maybe we need to show an error message if none of the date fields are focused?

> A more elegant fix would be to include a submit button

As I mentioned, this is a global component, so if we want this change, it should be easy to do, and it would be done globally.

**Comment by joshkimux** [comment]
@Mottie Thank you for the memory refresh and all the context! ❤️ 

> The "Day" select options are dependent on the "Month" selection, so there won't be any choices until a month is selected. It's not ideal, but this is the global date selector React component used across the site.

I'll write up a ticket for improving this. Would be great if we could adapt how [gov.uk handles dates.](https://design-system.service.gov.uk/components/date-input/)

> The anticipated date isn't fixed until the active service member gets closer to discharge, but we do need an exact date entered because they wouldn't be qualified for BDD (Benefits Delivery at Discharge) if they are not within the 90 to 180 days before that anticipated date. So we do need it to be as exact as possible, meaning we need the full date.

@kevinstachura I'm curious if research had insights on if folks:
- understood that anticipated date means all date fields are required and not just month/year
- were confident filling out all of the date fields

@mnorthuis , I'd love to get your thoughts on this too 🙇 

> Yes. If the incomplete date fields are blurred before a full date is entered, there is no feedback. It's done that way so the error messages don't show up while navigating between fields. Maybe we need to show an error message if none of the date fields are focused?

This is hard because we need to find a way to provide feedback at the right timing. I think showing the error message if none of the date fields makes sense since it won't bring back the issue we had before on moving backwards in a field.

> As I mentioned, this is a global component, so if we want this change, it should be easy to do, and it would be done globally.
🤞
